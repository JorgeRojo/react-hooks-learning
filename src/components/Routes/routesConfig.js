import Home from '../pages/Home/Home';
import Counter from '../pages/Counter/Counter';
import Api from '../pages/Api/Api';

const routesConfig = [
  {
    data: { menuOrder: 10, menuTitle: 'Counter' },
    path: '/counter',
    component: Counter,
    exact: false,
  },
  {
    data: { menuOrder: 0, menuTitle: 'Home' },
    path: '/',
    component: Home,
    exact: true,
  },
  {
    data: { menuOrder: 20, menuTitle: 'Api' },
    path: '/api',
    component: Api,
    exact: true,
  },
];

export default routesConfig;
