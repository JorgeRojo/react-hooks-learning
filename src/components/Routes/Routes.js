import React from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import routesConfig from './routesConfig';
import RouteNoMatch from '../pages/RouteNoMatch/RouteNoMatch';

const sortMenuItems = (prev, next) => prev.data.menuOrder - next.data.menuOrder;

const Routes = () => {
  return (
    <Router>
      <nav style={{ position: 'absolute' }}>
        <ul>
          {routesConfig
            .sort(sortMenuItems)
            .map(({ path, data: { menuTitle } }, index) => (
              <li key={index}>
                <Link to={path}>{menuTitle}</Link>
              </li>
            ))}
        </ul>
      </nav>

      <Switch>
        {routesConfig.map(({ exact, path, component }, index) => (
          <Route key={index} {...{ exact, path, component }} />
        ))}
        <Route path="*" component={RouteNoMatch} />
      </Switch>
    </Router>
  );
};

export default Routes;
