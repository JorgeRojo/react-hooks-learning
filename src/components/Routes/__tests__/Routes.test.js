import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import Routes from '../Routes';

describe('Routes Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Routes />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component', () => {
    const { container } = render(<Routes />);
    expect(container).toMatchSnapshot();
  });
});
