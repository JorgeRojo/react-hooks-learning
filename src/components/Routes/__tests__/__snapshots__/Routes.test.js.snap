// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component renders the component 1`] = `
<div>
  <nav
    style="position: absolute;"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/counter"
        >
          Counter
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="container text-center mt-4"
  >
    <h1
      class="display-4 mb-4"
    >
      Home
    </h1>
    <p>
      Testing react hooks
    </p>
  </div>
</div>
`;
